File: mobile/app/src/main/java/com/edvaldotsi/fastfood/model/ProdutoItem.java
package com.edvaldotsi.fastfood.model;

/**
 * Created by Edvaldo on 09/09/2015.
 */
public class ProdutoItem extends Produto {

    private static final int maxQuantidade = 3;

    private int quantidade;
    private boolean adicional;

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }

    public boolean isAdicional() {
        return adicional;
    }

    public void setAdicional(boolean adicional) {
        this.adicional = adicional;
    }

    public void incrementar() {
        if (quantidade < maxQuantidade) { quantidade++; }
    }

    public void decrementar() {
        if (quantidade > 0) { quantidade--; }
    }

    public float getValorTotal() {
        return quantidade * valor;
    }
}

File: fast_food/module/Application/Model/ProdutoItem.php
<?php

namespace Application\Model;

class ProdutoItem
{
    /**
     * @var Produto
     */
    private $produto;

    /**
     * @var Produto
     */
    private $item;

    /**
     * @var float
     */
    private $quantidade;

    /**
     * @var bool
     */
    private $adicional;

    /**
     * @var float
     */
    private $valor;

    /**
     * @return Produto
     */
    public function getProduto()
    {
        return $this->produto;
    }

    /**
     * @param Produto $produto
     */
    public function setProduto(Produto $produto)
    {
        $this->produto = $produto;
    }

    /**
     * @return Produto
     */
    public function getItem()
    {
        return $this->item;
    }

    /**
     * @param Produto $item
     */
    public function setItem(Produto $item)
    {
        $this->item = $item;
    }

    /**
     * @return float
     */
    public function getQuantidade()
    {
        return $this->quantidade;
    }

    /**
     * @param float $quantidade
     */
    public function setQuantidade($quantidade)
    {
        $this->quantidade = $quantidade;
    }

    /**
     * @return boolean
     */
    public function isAdicional()
    {
        return $this->adicional;
    }

    /**
     * @param boolean $adicional
     */
    public function setAdicional($adicional)
    {
        $this->adicional = $adicional;
    }

    /**
     * @return float
     */
    public function getValor()
    {
        return $this->valor;
    }

    /**
     * @param float $valor
     */
    public function setValor($valor)
    {
        $this->valor = $valor;
    }
}